core/txpool/legacypool/legacypool.go

```diff
@@ -585,7 +589,7 @@ func (pool *LegacyPool) validateTxBasics(tx *types.Transaction, local bool) erro
 		MaxSize: txMaxSize,
 		MinTip:  pool.gasTip.Load(),
 	}
-	if local {
+	if local && pool.config.PrioritizeLocals() {
```

Comment:
as far as I can tell, `local` will never be if `config.NoLocals` is set.  In your testcase, you call 

```
	if err := pool.addLocal(tx); !errors.Is(err, txpool.ErrUnderpriced) {
		t.Fatalf("Min tip not enforced")
	}

```
which invokes
```
func (pool *LegacyPool) addLocal(tx *types.Transaction) error {
	errs := pool.addLocals([]*types.Transaction{tx})
	return errs[0]
}

```
-> 
```
func (pool *LegacyPool) addLocals(txs []*types.Transaction) []error {
	return pool.addTxs(txs, !pool.config.NoLocals, true)
}
```
i.e `addTxs( txs, false, true ) `. So `local` is `false`. 

The test passes even without this change. Not sure what the original problem were that you tried to solve.

Reply:
@holiman, thanks for picking that up. The tests were missing checks. I lost them while reverting some changes suggested during the review that didn't pan out.

The issue I'm addressing is not with `addLocal`. The issue I'm addressing is with how local transactions are added using `Add` with local set to `true` after "the big refactor" (see @MariusVanDerWijden 's analysis  [here](https://github.com/ethereum/go-ethereum/pull/28435#pullrequestreview-1705640245)). 

tl;dr: right now, even if we set `txpool.nolocals`, `validateTxBasics` always sets `opts.MinTip` to `0` and `txpool.pricelimit` is not enforced. My PR fixes that.

Reply:
I see. I think I'd prefer doing the fix earlier: 
```diff
diff --git a/core/txpool/legacypool/legacypool.go b/core/txpool/legacypool/legacypool.go
index 0e33923274..853121421c 100644
--- a/core/txpool/legacypool/legacypool.go
+++ b/core/txpool/legacypool/legacypool.go
@@ -959,6 +959,7 @@ func (pool *LegacyPool) addRemoteSync(tx *types.Transaction) error {
 // If sync is set, the method will block until all internal maintenance related
 // to the add is finished. Only use this during tests for determinism!
 func (pool *LegacyPool) Add(txs []*types.Transaction, local, sync bool) []error {
+	local = local && !pool.config.NoLocals
 	// Filter out known ones without obtaining the pool lock or recovering signatures
 	var (
 		errs = make([]error, len(txs))

```

Reply:
@holiman thanks for the feedback. I've applied your suggestion